api/v1beta1/types.go

```diff
@@ -647,6 +647,10 @@ type APIServerLoadBalancer struct {
 	// specified.
 	// +optional
 	Provider optional.String `json:"provider,omitempty"`
+
+	// API Server LoadBalancer AvailabilityZone
+	// +optional
+	AvailabilityZone string `json:"availabilityZone,omitempty"`
```

Comment:
If AZ is optional it has to be `optional.String` which resolves to `*string`. This is a K8s API guideline.